### ISSUE TITLE
Paper Training Graph vs. Default Config

### ISSUE THREAD

**Issue by mithunvvv** [LucasCJYSDL/HierAIRL#2]
Hello,

Thank you once again for your paper! I was trying to recreate the results in the paper for Hopper with the default config file listed in the repo but am having some trouble understanding the training graph in the paper. For hopper, the default config is 2000 epochs of training to achieve a reward of 3500. In this image from your paper, the horizontal axis shows a large number of time steps. Is this the total number of time steps? If each episode is capped at a max of 1000 time steps (according to gym), how can we get 8 million time steps total if only running for 2000 epochs? Was the number of epochs greater than 2000 to recreate results? 

<img width="665" alt="Screenshot 2024-11-29 at 4 03 34 PM" src="https://github.com/user-attachments/assets/5a80d1f8-034c-4234-ba20-1c22c03be581">

**Comment by LucasCJYSDL** [comment]
If you check the config file, there are two parameters: "n_sample": 4096, "n_epoch": 2000, I think the x-axis is their product.

**Comment by mithunvvv** [comment]
Ok this makes sense thank you! :D So for Hopper for example, there are 1000 demos. 4096 refers to the number of transitions (ie.steps) we sample from the 1000 demos in a single epoch leading to 8 million epochs total. Thank you.

I tried to recreate the Hopper's results and am testing on the mujuco environment using gym's reward function. I am having some trouble recreating the convergence in the image above. I am noticing the reported reward of 3500 only when the mujuco environment reaches 1000 time steps. If the environment ends early (usually around 200-300 time steps) I don't see the reported reward of 3500 as in the paper (it is closer to a reward of 900-1000). Does the paper report the reward only considering the cases where the environment reaches 1000 time step. For example, these are some results from running `render_learned_model.py` 

```
Episode 103: R-sum: 1029.9761962890625; L-step: 334
Episode 104: R-sum: 3449.250732421875; L-step: 1000
Episode 105: R-sum: 1038.281005859375; L-step: 336
Episode 106: R-sum: 1022.5919189453125; L-step: 331
Episode 107: R-sum: 1022.8350830078125; L-step: 331
Episode 108: R-sum: 1035.6103515625; L-step: 336
Episode 109: R-sum: 1032.1849365234375; L-step: 335
Episode 110: R-sum: 1032.908935546875; L-step: 335
Episode 111: R-sum: 1023.9734497070312; L-step: 331
Episode 112: R-sum: 1036.6185302734375; L-step: 335
Episode 113: R-sum: 1035.309814453125; L-step: 336
Episode 114: R-sum: 1030.340576171875; L-step: 334
Episode 115: R-sum: 1036.860595703125; L-step: 335
Episode 116: R-sum: 1039.20068359375; L-step: 337
Episode 117: R-sum: 1041.94921875; L-step: 336
Episode 118: R-sum: 1040.968994140625; L-step: 337
Episode 119: R-sum: 1043.278076171875; L-step: 338
Episode 120: R-sum: 1043.0225830078125; L-step: 336
Episode 121: R-sum: 1020.9701538085938; L-step: 331
Episode 122: R-sum: 3429.078857421875; L-step: 1000
```

**Comment by LucasCJYSDL** [comment]
The trajectory return is the sum of rewards at each time step, so it would increase with the episode horizon.

**Comment by mithunvvv** [comment]
> The trajectory return is the sum of rewards at each time step, so it would increase with the episode horizon.

I am not sure I follow. For the total episodic reward, it is not converging after using the trained policy. for me It only gets around 3500 for the episodes that reach 1000 time steps.  That's why I am wondering if the reward in the plot is being showed for episodes of all time lengths or only those that reach 1000 time steps. Thank you for any clarification! :)

**Comment by LucasCJYSDL** [comment]
Let me check this from my side. For the paper, we didn't do the test that you have tried, for which more hyperparameter fine-tuning could be required.

**Comment by LucasCJYSDL** [comment]
The codebase is built on the paper: OptionGAIL (Adversarial Option-Aware Hierarchical Imitation Learning). In the caption of its Figure 3, they mentioned that they compared the **maximum** average reward-sums vs. exploration steps. I don't recall clearly, but the plots should be generated by testing the learned policy at each epoch for several trajectories and recording the maximum of these samples. We used the same metric as OptionGAIL since it's the main baseline.
